src/Compatibility/GenAPI/Microsoft.DotNet.GenAPI/CSharpFileBuilder.cs

```diff
@@ -272,7 +279,15 @@ private SyntaxNode GenerateAssemblyAttributes(IAssemblySymbol assembly, SyntaxNo
 
         private SyntaxNode GenerateForwardedTypeAssemblyAttributes(IAssemblySymbol assembly, SyntaxNode compilationUnit)
         {
-            foreach (INamedTypeSymbol symbol in assembly.GetForwardedTypes().Where(_symbolFilter.Include))
+            ImmutableArray<INamedTypeSymbol> forwardedTypesNamespaceSymbols = assembly.GetForwardedTypes();
+
+            // Support filtering out type forward attributes.
+            if (_attributeDataSymbolFilter is not null)
+            {
+                forwardedTypesNamespaceSymbols = forwardedTypesNamespaceSymbols.Where(_attributeDataSymbolFilter.Include).ToImmutableArray();
```

Comment:
Why are we filtering type-forwards for with the attribute filter?   That filter should only apply to the **application** of attributes.

Reply:
If anything, I'd expect typeforwards to be filtered with the Exclude API list, not the Exclude Attribute list.